openlibrary/catalog/add_book/__init__.py

```diff
@@ -908,11 +910,16 @@ def update_work_with_rec_data(
     """
     # Add subjects to work, if not already present
     if 'subjects' in rec:
-        work_subjects = list(work.get('subjects', []))
-        for s in rec['subjects']:
-            if s not in work_subjects:
-                work_subjects.append(s)
+        work_subjects: list[str] = list(work.get('subjects', []))
+        rec_subjects: list[str] = rec.get('subjects', [])
+        dedup_set = {subject.casefold() for subject in work_subjects}
+
+        for rec_subject in rec_subjects:
+            if rec_subject.casefold() not in dedup_set:
+                work_subjects.append(rec_subject)
+                dedup_set.add(rec_subject.casefold())
```

Comment:
We can use the `uniq` helper function for this

Reply:
Good call! I have so updated this PR.